sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/generators.scala

```diff
@@ -42,47 +42,27 @@ abstract class Generator extends Expression {
 
   override type EvaluatedType = TraversableOnce[Row]
 
-  override lazy val dataType =
-    ArrayType(StructType(output.map(a => StructField(a.name, a.dataType, a.nullable, a.metadata))))
+  override def dataType: DataType = ???
```

Comment:
As we moved the output field names from the `Generator` to `Generate`, probably it's impossible to construct the `StructType` any more within the `Generator`. And, it will be the rare case to the `dataType` method of a `Generator`, how about keep throwing exception if people do that?